### PR TITLE
Fixing APIs in 2023-02-01-preview

### PR DESCRIPTION
**Pull request by poshett**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (visingla-ms) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:1374]: There are a bunch of breaking changes reported on this PR. Please get a breaking changes approval from Azure Breaking Changes Reviewers <azbreakchangereview@microsoft.com> and ARM can review the changes afterwards. Once you have the approval, please remove the "ARMChangesRequested" label from the PR to make the PR visible to the ARM reviewers.

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (visingla-ms) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:68]: Pagination support?

**Comment by visingla-ms** [comment]
>           "description": "Marketplace Subscription Id",

NIT. The wrapper "marketplaceSubscription" should have description "Marketplace Subscription", not "Marketplace Subscription Id", right?

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2111 in 8c123f9. [](commit_id = 8c123f97a36320f4a42a236d03c7056ca08f213d, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "description": "Subscription Details: Marketplace SAAS Name",

Nit. What is "Subscription Details: " here?

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2121 in 8c123f9. [](commit_id = 8c123f97a36320f4a42a236d03c7056ca08f213d, deletion_comment = False)

**Comment by visingla-ms** [comment]
>         "emailId": {

Why nesting emailId perperty? Why not "UserEmailId" itself is modeled to be string?

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2408 in 8c123f9. [](commit_id = 8c123f97a36320f4a42a236d03c7056ca08f213d, deletion_comment = False)

**Comment by visingla-ms** [comment]
>     "UserEmailId": {

Small suggestion: you can name this parameter as "ApiKeyInputPayload" or something? That will help you to add more properties in addition to emailId if needed, in future.

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2404 in 8c123f9. [](commit_id = 8c123f97a36320f4a42a236d03c7056ca08f213d, deletion_comment = False)

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (visingla-ms) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:80]: > "in": "query", [](http://example.com/codeflow?start=11&length=15)

Parameter should have a description.

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (visingla-ms) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:86]: >"in": "query", [](http://example.com/codeflow?start=12&length=14)

Parameter should have a description.

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (visingla-ms) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2005]: >"ElasticVersionsListResponse": { [](http://example.com/codeflow?start=4&length=32)

Schema should have a description or title.

**Comment by visingla-ms** [comment]
There are some minor comments and small suggestions. Please check and remove "ARMChangesRequested" label after PR is ready for review again.

**Comment by poshett** [comment]
> > ```
> > "UserEmailId": {
> > ```
> 
> Small suggestion: you can name this parameter as "ApiKeyInputPayload" or something? That will help you to add more properties in addition to emailId if needed, in future.
> 
> Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2404 in [8c123f9](https://github.com/Azure/azure-rest-api-specs/commit/8c123f97a36320f4a42a236d03c7056ca08f213d). [](commit_id = [8c123f9](https://github.com/Azure/azure-rest-api-specs/commit/8c123f97a36320f4a42a236d03c7056ca08f213d), deletion_comment = False)

This API will not need any other input apart from EmailId from Users, hence this should suffice.

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (RheaHooda23) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:90]: The API only depends on the region parameter. I don't want to make it dependent on the subscription or tenant level. Let me know if that can work. Also, if region is the required parameter should I put it the api definition?

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (RheaHooda23) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:68]: No, max we will get 3 records

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (RheaHooda23) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:80]: Will add

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (RheaHooda23) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:86]: Will add

**Comment by RheaHooda23** [comment]
> > ```
> >       "description": "Marketplace Subscription Id",
> > ```
> 
> NIT. The wrapper "marketplaceSubscription" should have description "Marketplace Subscription", not "Marketplace Subscription Id", right?
> 
> Or if you will always have one property "Id" for this, avoid nesting and make proeprty as "marketplaceSubscriptionId".
> 
> Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2111 in [39cdb2d](https://github.com/Azure/azure-rest-api-specs/commit/39cdb2dd3e2053fa25a554713a759f63d9472711). [](commit_id = [39cdb2d](https://github.com/Azure/azure-rest-api-specs/commit/39cdb2dd3e2053fa25a554713a759f63d9472711), deletion_comment = False)

Sure I understand.

**Comment by RheaHooda23** [comment]
> > ```
> >       "description": "Subscription Details: Marketplace SAAS Name",
> > ```
> 
> Nit. What is "Subscription Details: " here?
> 
> Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2121 in [8c123f9](https://github.com/Azure/azure-rest-api-specs/commit/8c123f97a36320f4a42a236d03c7056ca08f213d). [](commit_id = [8c123f9](https://github.com/Azure/azure-rest-api-specs/commit/8c123f97a36320f4a42a236d03c7056ca08f213d), deletion_comment = False)

Marketplace subscription details

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (RheaHooda23) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2005]: Will add.

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:68]: >listElasticVersions [](http://example.com/codeflow?start=34&length=19)

should be a noun not a verb

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:83]: use odata filtering instead with $filter query param. The rest api specs repo should have some examples

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:89]: >Subscription ID where monitor resource will be deployed [](http://example.com/codeflow?start=28&length=55)

Subscription ID where monitor resource is deployed

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:92]: Whats the reasoning for this given the actual resources are under a subscription?

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:1420]: >getOrganizationApiKey [](http://example.com/codeflow?start=65&length=21)

if u call this listOrganizationApiKey, you can let your customers use this in ARM templates

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2014]: List response schema should have a"value" property and the actual array should be inside value

Comment (rkmanda) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2014]: along with a nextLink. Look at other examples in the same file

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:68]: /providers/Microsoft.Elastic/elasticVersions

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (poshett) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:1420]: @rkmanda, did you mean to change the API name from "getOrganizationApiKey" to "listOrganizationApiKey" ?
Apologies, the comment is a tad confusing.

Comment (poshett) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json]: If yes, updated and renamed API as suggested in https://github.com/Azure/azure-rest-api-specs/pull/23343/commits/bfd1a05f2352a3b650b258f1ae2428a531036318

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (RheaHooda23) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2014]: At max, we will get 10 records. Not more than that.

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (RheaHooda23) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:83]: This is just a regional API that fetches some elastic versions on the basis of region. Scope of this is limited to region only. I don't think we would need to use odata filtering for this at least.

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (RheaHooda23) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:68]: For this API path, "https://management.azure.com/providers/Microsoft.Elastic/elasticVersions", would I need to do anything other than manifest rollout for it to function.

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (RheaHooda23) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:89]: Updated

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json]: I dont follow the connection between my comment nd your question

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:83]: odata filters are a standard way to specify how the results need to be filtered. It does not matter whether there is one criteria or several. Can u provide a rationale for why you cannot use OData filters?

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:92]: Can u provide an answer for this?

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:2014]: ok. please keep in mind that anytime in future if you would want to change this to a pageable result it will be flagged as a breaking change and will need you to get a breaking changes approval.

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:110]: >list [](http://example.com/codeflow?start=65&length=4)

when its a GET operation, you MUST use a noun like elasticVersions

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:129]: >region [](http://example.com/codeflow?start=21&length=6)

still not using odata?

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [inline specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:1802]: >generateApiKey [](http://example.com/codeflow?start=9&length=14)

consider making it an enum

**Review by ihsanali01 — APPROVED** [review]
Thanks & Go ahead please.

**Review by ihsanali01 — APPROVED** [review]
Thanks & Go ahead please.

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (poshett) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:1802]: It is just a true/false value hence boolean seems more appropriate.

**Review by ihsanali01 — APPROVED** [review]

File: specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json
Comment (rkmanda) [reply specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-02-01-preview/elastic.json:129]: Please fix this in the next preview or GA API version

**Comment by RheaHooda23** [comment]
Adding the work item for some improvements in the next swagger version: https://msazure.visualstudio.com/Liftr/_sprints/backlog/Liftr-Elastic/Liftr/Elastic/Sprint%200523?workitem=17964014

**Comment by XiaofeiCao** [comment]
@tadelesh @msyyc Service has moved `subscriptionId` from client to method. 

Please help review the SDK breaking change on Go and Python. Thanks!

**Comment by tadelesh** [comment]
> @tadelesh @msyyc Service has moved `subscriptionId` from client to method.
> 
> Please help review the SDK breaking change on Go and Python. Thanks!

It is uncommon to make the subscriptionId as the method parameter. All service's operations contains this param. Is it an intentional change?

**Comment by XiaofeiCao** [comment]
> It is uncommon to make the subscriptionId as the method parameter. All service's operations contains this param. Is it an intentional change?

Thanks @tadelesh .

@poshett May I know why this PR changes the parameter location of `subscriptionId`?

**Comment by tadelesh** [comment]
Also, if it is intentional, has the breaking change review board approve it?

**Comment by RheaHooda23** [comment]
> Also, if it is intentional, has the breaking change review board approve it?

Added it mistakenly, have reverted. Thanks for pointing that out.

**Comment by RheaHooda23** [comment]
> Also, if it is intentional, has the breaking change review board approve it?

Added it mistakenly, have reverted. Thanks for pointing that out.

**Comment by XiaofeiCao** [comment]
Removing `CI-BreakingChange-Python-Track2` since no breaking change for python now.

**Comment by XiaofeiCao** [comment]
Thanks @tadelesh for the help!

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**
